devito/operator/symbols.py

```diff
@@ -0,0 +1,33 @@
+from devito.tools import generator
+
+__init__ = ['SymbolRegistry']
+
+
+class SymbolRegistry(object):
+
+    """A registry for all the symbols used by an Operator."""
+
+    _symbol_prefix = 'r'
+
+    def __init__(self, nthreads=None, nthreads_nested=None, nthreads_nonaffine=None,
+                 threadid=None):
+        # {name -> generator()} -- to create unique names for symbols, functions, ...
+        self.counters = {}
+
+        # Special symbols
+        self.nthreads = nthreads
+        self.nthreads_nested = nthreads_nested
+        self.nthreads_nonaffine = nthreads_nonaffine
+        self.threadid = threadid
+
+    def make_name(self, prefix=None):
```

Comment:
nitpicking: I know was somewhere but can't find it. It does make a name but sounds like it's creating the name of the SymbolRegistery. Maybe something like "next_registered_symbol" ?

Reply:
it may sound like, but it's not a classmethod, so it can't be that. I mean, one of the two main points of the SymbolRegistry is to create unique names, so...